lib/lightning/runs/run.ex

```diff
@@ -132,15 +141,17 @@ defmodule Lightning.Run do
   end
 
   @doc """
-  Adds options (project-level logging options, resource limits such as timeout
+  Adds options (workflow-level and project-level logging options, resource limits such as timeout
   and memory usage, etc.) to the run before storing in the DB.
   """
-  def add_options(changeset, project_id) do
+  def add_options(changeset, workflow_id, project_id) do
+    workflow = Workflows.get_workflow!(workflow_id)
+
     put_change(
       changeset,
       :options,
       UsageLimiter.get_run_options(%Context{project_id: project_id})
-      |> Enum.into(%{})
+      |> Enum.into(%{enable_job_logs: workflow.enable_job_logs})
     )
   end
```

Comment:
Looking at this again, I can see the temptation to put this function in here (it's working with changesets), but since it calls out to the usage limiter; I would argue this function should be moved to the WorkOrders or Runs module.